### PR TITLE
Features/add-logging-action

### PR DESCRIPTION
**Pull request by BRuslanB**
All activity records are stored in the app.log file
Calls and actions of all methods are logged, including parameter values.
Also added is logging of actions related to user registration, login, logout and password change.

### CODE REVIEW

**Comment by BRuslanB** [comment]
In the second commit, the function of hiding characters for user passwords has been added.

File: src/main/java/kz/bars/familybudget/api/BudgetController.java [hunk 1]
```diff
@@ -16,24 +17,27 @@
...
     @GetMapping(value = "dates/{date1}/{date2}")
-    @Operation(description = "Получение списка доходов за период с.. по..")
-    public List<BudgetDto> getAllBudgetBetweenDate(@Parameter(description = "дата начала")
+    @Operation(description = "Getting a list of Income for the period from.. to..")
+    public List<BudgetDto> getAllBudgetBetweenDate(@Parameter(description = "date from")
                                                    @PathVariable(name = "date1") LocalDate dateFrom,
-                                                   @Parameter(description = "дата окончания")
+                                                   @Parameter(description = "date to")
                                                    @PathVariable(name = "date2") LocalDate dateTo) {
+        log.info("!Getting a list of Income for the period " + "from " + dateFrom + " to "+ dateTo);
```
Comment (btimur) [inline src/main/java/kz/bars/familybudget/api/BudgetController.java:40]: log.info("!Getting a list of Income for the period from [{}]  to [{}] ", dateFrom, dateTo);

File: src/main/java/kz/bars/familybudget/controller/HomeController.java [hunk 2]
```diff
@@ -155,10 +191,11 @@ public String deleteExpenseCategory(@RequestParam(name = "expense_category_id")
 
         try {
             expenseCategoryService.deleteExpenseCategory(expenseCategoryId);
+            log.info("!Category removed, " + "id=" + expenseCategoryId);
             return "redirect:/settings?category_success";
 
         } catch (Exception e) {
-
+            log.info("!Category not removed, " + "id=" + expenseCategoryId);
```
Comment (btimur) [inline src/main/java/kz/bars/familybudget/controller/HomeController.java:198]: log.error("!Category not removed, " + "id=" + expenseCategoryId), e)

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -41,25 +39,18 @@ management:
...
-#            token-uri: https://oauth2.googleapis.com/token
-#            user-info-uri: https://www.googleapis.com/oauth2/v3/userinfo
-#            user-name-attribute: email
\ No newline at end of file
+# logging setting
+logging:
+  file:
+    name: logs/app.log
+    append: true
+  loggers:
```
Comment (btimur) [inline src/main/resources/application.yml:48]: root:
 level : ERROR

**Review by btimur — APPROVED** [review]

File: src/main/resources/application.yml (hunk 3 above)
Comment (BRuslanB) [reply src/main/resources/application.yml:50]: added this recomendation

File: src/main/java/kz/bars/familybudget/api/BudgetController.java (hunk 1 above)
Comment (BRuslanB) [reply src/main/java/kz/bars/familybudget/api/BudgetController.java:40]: done fix

File: src/main/java/kz/bars/familybudget/controller/HomeController.java (hunk 2 above)
Comment (BRuslanB) [reply src/main/java/kz/bars/familybudget/controller/HomeController.java:198]: done

**Pull request merged**